src/main/webapp/app/exercises/programming/shared/service/external-cloning.service.ts

```diff
@@ -12,26 +13,16 @@ export class ExternalCloningService {
     }
 
     /**
-     * Builds an url to clone a Repository in IntelliJ
-     * Structure: jetbrains://idea/checkout/git?idea.required.plugins.id=Git4Idea&checkout.repo=:RepoUrl
+     * Builds an url to clone a Repository in the corresponding ide
      * @param cloneUrl the url of the repository to clone
+     * @param ide the ide with the deeplink to build the url for
      */
-    buildJetbrainsUrl(cloneUrl: string | undefined): string | undefined {
+    // TODO shift exercise Programming Language Logic here
```

Comment:
Yes that would be nice, currently the functionality to get and pass the language is split over 3 files 😆

Reply:
to keep the service stateless, i will not move the logic there